### PR TITLE
Multiple data offset

### PR DESCRIPTION
**Pull request by vkter**
Offset Multiple Data
image

### CODE REVIEW

**Comment by Durman** [comment]
image

File: nodes/list_struct/shift_mk2.py [hunk 1]
```diff
@@ -102,6 +110,8 @@ def process(self):
                         d.rotate(number)
                         sub_output.append(list(d))
                     output.append(sub_output)
+        return output
+
 
 
         self.outputs['data'].sv_set(output)
```
Comment (Durman) [inline nodes/list_struct/shift_mk2.py:117]: I guess last line can be removed.

File: nodes/list_struct/shift_mk2.py [hunk 2]
```diff
@@ -66,8 +66,16 @@ def process(self):
...
+        number = self.inputs["shift"].sv_get(deepcopy=False)
+        output = []
+        for d,n in zip(data,number):
+            out = self.new_method(d, n[0])
+            output.append(out)
 
+
+        self.outputs['data'].sv_set(output)
+
+    def new_method(self, data, number):
```
Comment (Durman) [inline nodes/list_struct/shift_mk2.py:78]: The name of the new method should be something more meaningful. Something like roll list.

**Comment by vkter** [comment]
Whether the applications can be combined？

**Comment by Durman** [comment]
Could you add any description?

**Comment by vkter** [comment]
Offset Multiple Data
image

**Comment by Durman** [comment]
If you add vectorization to one input I guess it's better to add it to another input socket as well. In the example below I would expect two lists to be created by the node.

image

Also it's not clear how to make it work consistently with work of other similar nodes. For example List Slice node has on object in each socket but generate multiple ones.

image

**Comment by vkter** [comment]
The internal code uses zip () to match the input data. In short, it matches the data according to the shortest, which seems to be no problem

**Comment by Durman** [comment]
Nevermind, it can be fixed later.

**Pull request merged**